### PR TITLE
Add helpers to encourage migration

### PR DESCRIPTION
**Pull request by ndarilek**
I extracted the messaging from `fly deploy` to a separate function so we can a) call it from more places and b) change how assertive/aggressive it gets over time (E.g. maybe we eventually offer to run the migration automatically on interactive terminals prior to forcing it.)

I also called this function from `fly status`. I don't have any Nomad apps to test with but I inverted the check to make sure it worked, and it seemed to. I think I uninverted the invert but definitely check since I'm unfortunately out of coffee today so who knows what my brain is doing to get back at me for that.

### CODE REVIEW

**Review by alichay — APPROVED** [review]
nice!

**Review by alichay — APPROVED** [review]
nice!

**Comment by ndarilek** [comment]
I've been told that `AppCompact` is the only reliable source of which platform an app is on.

Is it correct that this can only be retrieved from an API call? Or does this get put in the context at some point? I'm still not entirely clear what ends up in the context, what has to be retrieved from the API, or whether retrieving something from the context performs background magic like API calls or filesystem reads.

Thanks.

**Comment by alichay** [comment]
> I've been told that `AppCompact` is the only reliable source of which platform an app is on.
> 

yeah I just saw that! that's totally right. the places where you call the new function should have this info _anyway_, so if you just make the helper take an appcompact, that is probably the path of least resistance?

**Comment by ndarilek** [comment]
Awesome, thanks for the feedback. I think this should do it, and also answers my questions about whether or not this makes sense to add to non-app commands.

File: internal/command/apps/restart.go [hunk 1]
```diff
@@ -72,9 +72,14 @@ func runRestart(ctx context.Context) error {
...
 
 func runNomadRestart(ctx context.Context, app *api.AppCompact) error {
-	command.PromptToMigrate(ctx)
 	client := client.FromContext(ctx).API()
 
+	appName := flag.FirstArg(ctx)
+	app, err := client.GetAppCompact(ctx, appName)
+	if err != nil {
+		command.PromptToMigrate(ctx, app)
+	}
```
Comment (alichay) [inline internal/command/apps/restart.go:81]: Getting the app is unnecessary here, we already have one passed as an argument to the function :)
```suggestion
	command.PromptToMigrate(ctx, app)
```

**Comment by ndarilek** [comment]
Ugh, and once again with the tire fire that is GitHub Actions accessibility.

I see there are lint failures, but I can't find where they are with my screen reader. Further, when I use my usual trick of checking the raw logs, I get this:

```
2023-05-18T17:57:09.1055385Z Running [/home/runner/golangci-lint-1.52.2-linux-amd64/golangci-lint run --out-format=github-actions] in [] ...
2023-05-18T17:57:56.5540969Z ##[error]SA4009: argument app is overwritten before first use (staticcheck)
2023-05-18T17:57:56.5551838Z ##[error]SA4009(related information): assignment to app (staticcheck)
2023-05-18T17:57:56.5552434Z 
2023-05-18T17:57:56.5552977Z ##[error]issues found
2023-05-18T17:57:56.5553790Z Ran golangci-lint in 47447ms
```

Am I just missing the file/line number of the error? I imagine it's somewhere in that giant log but I don't have the rest of eternity to review it line-by-line. :)

Thanks.

**Comment by alichay** [comment]
> Ugh, and once again with the tire fire that is GitHub Actions accessibility.
> 
> I see there are lint failures, but I can't find where they are with my screen reader. Further, when I use my usual trick of checking the raw logs, I get this:
> 
> ```
> 2023-05-18T17:57:09.1055385Z Running [/home/runner/golangci-lint-1.52.2-linux-amd64/golangci-lint run --out-format=github-actions] in [] ...
> 2023-05-18T17:57:56.5540969Z ##[error]SA4009: argument app is overwritten before first use (staticcheck)
> 2023-05-18T17:57:56.5551838Z ##[error]SA4009(related information): assignment to app (staticcheck)
> 2023-05-18T17:57:56.5552434Z 
> 2023-05-18T17:57:56.5552977Z ##[error]issues found
> 2023-05-18T17:57:56.5553790Z Ran golangci-lint in 47447ms
> ```
> 
> Am I just missing the file/line number of the error? I imagine it's somewhere in that giant log but I don't have the rest of eternity to review it line-by-line. :)
> 
> Thanks.

The issue it's complaining about is the thing I mentioned in the comment above. it just doesn't like that app is re-assigned

**Comment by ndarilek** [comment]
Ah, I didn't know where that particular action vomited its report. Thanks, I'll check there in the future in case that is any more accessible.

**Comment by ndarilek** [comment]
Also, is there any way to run these lints locally? I wish VS Code's problems view surfaced them somehow, since I generally check that before pushing code.

**Review by alichay — APPROVED** [review]

**Pull request merged**